### DISCUSSION TITLE
zstd compressed export option

### DISCUSSION THREAD

**Discussion by Be-ing** [microsoft/vcpkg#17927] [Ideas]
**Is your feature request related to a problem? Please describe.**
Large ZIP files from `vcpkg export` can take a long time to decompress. For our 1.0 GB ZIP file, it takes several minutes to decompress on GitHub Actions.

**Proposed solution**
Add an option to `vcpkg export` to export to a zstd compressed tar archive.

**Describe alternatives you've considered**
I'm not sure if any other archive format supports zstd, but that may be an alternative if it exists.

**Additional context**
[![bar graph demonstrating zstd decompression is far faster than other compression algorithms](https://raw.githubusercontent.com/facebook/zstd/v1.3.4/doc/images/DSpeed3.png)](https://facebook.github.io/zstd/)

**Comment by StarGate-One** [comment]
Have you tried the `--7zip` option, although I do not know if `GitHub Actions` can consume a .7z file?
>PS D:\2019\test> ./vcpkg help export
Example:
  vcpkg export zlib zlib:x64-windows boost --nuget

>Options:
  --dry-run                       Do not actually export
  --raw                           Export to an uncompressed directory
  --nuget                         Export a NuGet package
  --ifw                           Export to an IFW-based installer
  --zip                           Export to a zip file
 **--7zip                          Export to a 7zip (.7z) file**
  --x-chocolatey                  Export a Chocolatey package (experimental feature)
  --prefab                        Export to Prefab format
  --prefab-maven                  Enable maven
  --prefab-debug                  Enable prefab debug
  --x-all-installed               Export all installed packages
  --output=...                    Specify the output name (used to construct filename)
  --output-dir=...                Specify the output directory for produced artifacts
  --nuget-id=...                  Specify the id for the exported NuGet package (overrides --output)
  --nuget-description=...         Specify a description for the exported NuGet package
  --nuget-version=...             Specify the version for the exported NuGet package
  --ifw-repository-url=...        Specify the remote repository URL for the online installer
  --ifw-packages-directory-path=...
                                  Specify the temporary directory path for the repacked packages
  --ifw-repository-directory-path=...
                                  Specify the directory path for the exported repository
  --ifw-configuration-file-path=...
                                  Specify the temporary file path for the installer configuration
  --ifw-installer-file-path=...   Specify the file path for the exported installer
  --x-maintainer=...              Specify the maintainer for the exported Chocolatey package
                                  (experimental feature)
  --x-version-suffix=...          Specify the version suffix to add for the exported Chocolatey
                                  package (experimental feature)
  --prefab-group-id=...           GroupId uniquely identifies your project according maven
                                  specifications
  --prefab-artifact-id=...        Artifact Id is the name of the project according maven

**Comment by Be-ing** [comment]
As shown in the graph above, lzma decompression is slower than zlib. zstd is by far faster than both.

**Comment by JackBoosY** [comment]
cc @ras0219-msft

**Comment by netcorefan1** [comment]
May I suggests something like `%VCPKG_BINARY_CACHE_COMPRESSION%  (LZMA2 | zlib | etc)` ?
As of now, binary cached files are saved as [.zip](https://github.com/microsoft/vcpkg/discussions/30637) files which is an useless waste of disk space.
